### PR TITLE
Added support for package-private jsonAdapter method (#87)

### PR DESCRIPTION
**Pull request by nedtwigg**

### CODE REVIEW

File: auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java [hunk 1]
```diff
@@ -102,14 +102,13 @@ boolean nullable() {
...
-    // check that the class contains a public static method returning a JsonAdapter
+    // check that the class contains a static method returning a JsonAdapter
     TypeElement type = context.autoValueClass();
     ParameterizedTypeName jsonAdapterType = ParameterizedTypeName.get(
         ADAPTER_CLASS_NAME, TypeName.get(type.asType()));
     TypeName returnedJsonAdapter = null;
     for (ExecutableElement method : ElementFilter.methodsIn(type.getEnclosedElements())) {
-      if (method.getModifiers().contains(Modifier.STATIC)
-          && method.getModifiers().contains(Modifier.PUBLIC)) {
+      if (method.getModifiers().contains(Modifier.STATIC)) {
```
Comment (grandstaish) [inline auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java]: Minor: you could also check it's non-private here

**Comment by nedtwigg** [comment]
Seems like the simplest thing to implement, and the easiest thing to
communicate to users, is just "have a static method".

Is there a reason to disallow private static?

**Comment by ZacSweers** [comment]
> Is there a reason to disallow private static?

I think the point was more that we can't access private methods from the generated factory, so you should only count non-private ones

File: auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java (hunk 1 above)
Comment (ZacSweers) [reply auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java]: Agreed, we need to check this

**Review by ZacSweers — CHANGES_REQUESTED** [review]

**Comment by nedtwigg** [comment]
Fair enough.  Plain-old javac will give the user those warnings, but it's easy to add a little complexity to check it here as well.

**Comment by ZacSweers** [comment]
> in-old javac will give the user those warnings

I think the point is they should be allowed to have a private static method if they want without warnings, and the processor should skip those.

**Comment by nedtwigg** [comment]
Okey doke :)

**Comment by ZacSweers** [comment]
Just to go last mile here - we could just check the package and make sure it's accessible, and then ignore anything not accessible. That's what we do in RAVE [here](https://github.com/uber-common/rave/blob/master/rave-compiler/src/main/java/com/uber/rave/compiler/RaveProcessor.java#L153-L154)

**Comment by nedtwigg** [comment]
IMO, the error-checking in this code is harder to read than what the code itself is doing, and it is redundant with what the javac compiler does for free.

Adding a private check was easy, but adding this check is harder enough that my initiative on this minor task is exhausted.  Making our jsonAdapter methods public isn't a huge deal, we'll just continue to workaround in that way.  Feel free to close 👍

**Comment by ZacSweers** [comment]
Its a few more lines and I think a processor should avoid generating invalid code if it can avoid it. I can just do it in another PR if you don't want to then

**Comment by ZacSweers** [comment]
It also prevents users from intentionally keeping methods exposed to certain packages

**Pull request closed without merging**